File: Python/Baekjoon/.1000/1644.py
N = int(input())
prime = [True for i in range(N + 1)]
prime[0] = prime[1] = False

for i in range(2, int(N**0.5) + 1):
    if not prime[i]:
        continue
    for j in range(i * 2, N + 1, i):
        prime[j] = False

sum = [0]
for i in range(N + 1):
    if prime[i]:
        sum.append(sum[-1] + i)

i, j = 0, 1
res = 0
while j < len(sum):
    if sum[j] - sum[i] > N:
        i += 1
    elif sum[j] - sum[i] < N:
        j += 1
    else:
        res += 1
        j += 1
print(res)

File: PS/Baekjoon/.1000/1644.cpp
#include <bits/stdc++.h>
#define int long long

using namespace std;
typedef pair<int, int> pr;
typedef tuple<int, int, int> tp;

vector<bool> get_primes(int N)
{
    vector<bool> primes(N + 1);
    fill(primes.begin(), primes.end(), true);
    primes[0] = false, primes[1] = false;

    for (int i = 2; i * i <= N; i++)
    {
        if (!primes[i])
            continue;
        for (int j = i * 2; j <= N; j += i)
            primes[j] = false;
    }

    return primes;
}

signed main()
{

    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);

    int N;
    cin >> N;

    vector<bool> primes = get_primes(N);
    vector<int> sum;
    sum.push_back(0);

    for (int i = 1; i <= N; i++)
    {
        if (primes[i])
            sum.push_back(sum.back() + i);
    }

    int i = 0, j = 1, res = 0;

    while (j < sum.size())
    {
        if (sum[j] - sum[i] > N)
            i++;
        else if (sum[j] - sum[i] < N)
            j++;
        else
        {
            res++;
            j++;
        }
    }

    cout << res;

    return 0;
}